src/types/rule.cpp

```diff
@@ -335,7 +352,7 @@ void Rule::setLocation(const std::string& filePath, std::uint64_t lineNumber)
  */
 bool Rule::isGlobal() const
 {
-	return _mod.has_value() && (*_mod)->getType() == TokenType::GLOBAL;
+	return _mod_global.has_value();
```

Comment:
This code is correct, but it expects, that when the rule is not global, the `_mod_global` is `std::nullopt`. The above change in the `Rule` constructor is therefore necessary.